### ISSUE TITLE
The cross attention masking is disabled

### ISSUE THREAD

**Issue by mmehdig** [pytorch/torchtune#2552]
It seems the cross-attention masking is disabled here:
https://github.com/pytorch/torchtune/blob/3fc4c95bef9911da907d9403c9196c024a42089f/torchtune/modules/transformer.py#L284

I added a comment/question on the old pr:
https://github.com/pytorch/torchtune/pull/1548#discussion_r2027406709

I wonder if it is still needed? 

If this is about a specific SDPBackend, perhaps it should be documented in that direction?

**Comment by joecummings** [comment]
@pbontrager Can you weigh in here?

**Comment by pbontrager** [comment]
This is disabling the mask for all positions (tokens) where there are no images to cross attend to. A common example is the first text tokens that are before any images. All of the image embeddings would be masked out which would cause the softmax to NaN. What happens here is that we turn off the mask for that row and then 0 out those attention values [here](https://github.com/pytorch/torchtune/blob/3fc4c95bef9911da907d9403c9196c024a42089f/torchtune/modules/transformer.py#L292).

Now, as the comment says, since PyTorch 2.5 the softmax layer can output 0s instead of NaNs when a row is fully masked out. This would allow us to remove the entire concept of "skip_mask" from the code. This would likely make the layer a bit more efficient and cleaner, but would have to be tested to verify it's the same.
